In [1]:
#imports
import pandas as pd
import numpy as np
import re
from sklearn.feature_extraction.text import CountVectorizer

import warnings
warnings.filterwarnings('ignore')

In [2]:
vg = pd.read_csv('../data/video_games.csv')

In [3]:
vg.head()

customer_id       review_id  product_id  product_parent  \
0     12039526   RTIS3L2M1F5SM  B001CXYMFS       737716809   
1      2331478  R3BH071QLH8QMC  B0029CSOD2        98937668   
2     52495923  R127K9NTSXA2YH  B00GOOSV98        23143350   
3     14533949  R32ZWUXDJPW27Q  B00Y074JOM       821342511   
4     17521011  R2F0POU5K6F73F  B008XHCLFO        24234603   

                                       product_title  star_rating  \
0         Thrustmaster T-Flight Hotas X Flight Stick            5   
1  Hidden Mysteries: Titanic Secrets of the Fatef...            1   
2  GelTabz Performance Thumb Grips - PlayStation ...            3   
3  Zero Suit Samus amiibo - Japan Import (Super S...            4   
4                         Protection for your 3DS XL            5   

   helpful_votes  total_votes  verified_purchase review_date  \
0              0            0                  1  2015-08-31   
1              0            1                  1  2015-08-31   
2              0            0                  1  2015-08-31   
3              0            0                  1  2015-08-31   
4              0            0                  1  2015-08-31   

                                         full_review  
0  amazing joystick I especially love twist Used ...  
1              One Star poor quality work advertised  
2  good could bettee nice tend slip away stick in...  
3  Great flawed Great amiibo great collecting Qua...  
4  A Must I 2012 2013 XL durable comfortable real...

In [5]:
vg['star_rating'].value_counts(normalize=True)

5    0.580267
4    0.178084
1    0.104341
3    0.085006
2    0.052302
Name: star_rating, dtype: float64

For this problem - baseline accuracy will be 58% (predicting all reviews as 5 star). My goal is to exceed that through NLP.